doc/README.rst

```diff
@@ -0,0 +1,83 @@
+The doc folder
+===============
+
+The documentation for NEST is contained in this folder. We are currently working
+on improving the documentation and making updates. However, if you notice
+something out of place, you can `submit an issue <https://nest.github.io/nest-simulator/development_workflow#reporting-bugs-and-issues>`_
+in our GitHub repository.
+
+For **developer documentation**, we use `doxygen <http://doxygen.org/>`__
+comments extensively throughout NEST.
+After installing NEST, you can extract comments from the source code with
+``make doc`` and a doxygen folder with html files will be generated in the doc
+folder in your source directory.
+
+For a list of commands for SLI and C++, you can access the the online command
+index via the command line
+
+::
+
+   import nest
+   nest.helpdesk()
+
+
+.. note::
+
+ The command ``helpdesk()`` needs to know which browser to launch in order to display
+ the help pages. The browser is set as an option of helpdesk. Please see the file
+ ``~/.nestrc`` for an example setting firefox as browser.
+ Please note that the command helpdesk does not work if you have compiled
+ NEST with MPI support, but you have to enter the address of the helpdesk
+ (file:///install/path/share/doc/nest/index.html) manually into the browser.
+ Please replace ``/install/path`` with the path under which NEST is installed.
+
+Local install of user documentation
+--------------------------------------
+
+To access user documentation, you can check out the website or generate
+the html files by following the steps below.
+
+Requirements
+~~~~~~~~~~~~
+
+To keep everything simple and easy to maintain, the installation with
+conda is described here. So all you need is conda. If you don’t have it
+yet, `miniconda <https://conda.io/miniconda.html>`__ is recommended.
+
+Installation
+~~~~~~~~~~~~
+
+With the following steps a full development environment is installed -
+even NEST 2.14.0.
+
+Change to the doc/ folder (if you’re not already there).
+
+::
+
+   cd ./doc
+
+Create and activate the environment.
+
+::
+
+   conda env create -f environment.yml
```

Comment:
I tried this on my computer (with full Anaconda installed) and go the following error:
```
ResolvePackageNotFound: 
  - nest
  - libstdcxx-ng
  - libgcc-ng
```

Here is my conda configuration:
```
     active environment : None
          conda version : 4.5.11
    conda-build version : 3.15.1
         python version : 3.6.6.final.0
       base environment : /Users/plesser/anaconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/osx-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : $HOME/anaconda3/pkgs
                          $HOME/.conda/pkgs
       envs directories : $HOME/anaconda3/envs
                          $HOME/.conda/envs
               platform : osx-64
             user-agent : conda/4.5.11 requests/2.19.1 CPython/3.6.6 Darwin/17.7.0 OSX/10.13.6
             netrc file : None
           offline mode : False
```

Reply:
I cannot reproduce this error in Ubuntu. It seems to be something MacOS specific.

Reply:
From 
- https://anaconda.org/anaconda/libgcc-ng
- https://anaconda.org/anaconda/libstdcxx-ng
it appears that these packages are available for Linux only.  Could they be marked as optional/linux-specific in the environment file?

The same holds for NEST, see
- https://bioconda.github.io/recipes.html

Until we have a Bioconda recipe for OSX as well, the doc should state that this approach works for Linux only and provide instructions for "manual" installation as well.